kafka/producer.go

```diff
@@ -99,16 +118,23 @@ func (p *Producer) produce(msg *Message, msgFlags int, deliveryChan chan Event)
 		cgoid = p.handle.cgoPut(cgoDr{deliveryChan: deliveryChan, opaque: msg.Opaque})
 	}
 
-	r := int(C.do_produce(crkt, C.int32_t(msg.TopicPartition.Partition),
+	var timestamp int64
+	if !msg.Timestamp.IsZero() {
```

Comment:
Fun fact: Go's `time` default value is not 1970, but 0001-01-01..
Very convenient.